File: src/go-backend/internal/models/episode.go
package models

import "time"

// URL params
type EpisodeRequest struct {
	ServiceTag    string
	ShowID        string
	SeasonNumber  int
	EpisodeNumber int
}

// Episode Response
type Episode struct {
	Streams            *[]EpisodeStream    `json:"streams"`
	BackdropURL        string              `json:"backdropURL"`
	Number             int                 `json:"number"`
	Name               string              `json:"name"`
	OriginalName       string              `json:"originalName"`
	Overview           string              `json:"overview"`
	PosterURL          string              `json:"posterURL"`
	MediaType          string              `json:"mediaType"`
	OriginalLanguage   string              `json:"originalLanguage"`
	Length             float64             `json:"length"`
	Cast               []string            `json:"cast"`
	Directors          []string            `json:"directors"`
	FirstAirDate       time.Time           `json:"firstAirDate"`
	OriginCountry      string              `json:"originCountry"`
	AvailabilityStatus string              `json:"availabilityStatus"`
	Described          bool                `json:"described"`
	VideoTracks        []EpisodeVideoTrack `json:"videoTracks"`
	AudioTracks        []EpisodeAudioTrack `json:"audioTracks"`
	TextTracks         []EpisodeTextTrack  `json:"textTracks"`
	CuePoints          []EpisodeCuePoint   `json:"cuePoints"`
}

// Next Episode Response
type NextEpisode struct {
	ShowID       string `json:"showID"`
	SeasonNumber int    `json:"seasonNumber"`

	Episode
}

type EpisodeStream struct {
	Type string `json:"type"`
	URL  string `json:"url"`
}

type EpisodeVideoTrack struct {
	Name    string `json:"name"`
	Height  int    `json:"height"`
	Width   int    `json:"width"`
	Bitrate int    `json:"bitrate"`
}

type EpisodeAudioTrack struct {
	Code         string `json:"code"`
	Name         string `json:"name"`
	OriginalName string `json:"originalName"`
}

type EpisodeTextTrack struct {
	Type     string `json:"type"`
	Name     string `json:"name"`
	Language string `json:"language"`
	TrackURL string `json:"trackURL"`
}

type EpisodeCuePoint struct {
	Name  string  `json:"name"`
	Start float64 `json:"start"`
	End   float64 `json:"end"`
}

File: src/frontend/src/lib/types/episode.ts
export type EpisodeRequest = {
    ServiceTag: string;
    ShowID: string;
    SeasonNumber: number;
    EpisodeNumber: number;
};

export type Episode = {
    streams: EpisodeStream[];
    backdropURL: string;
    number: number;
    name: string;
    originalName: string;
    overview: string;
    posterURL: string;
    mediaType: string;
    originalLanguage: string;
    length: number;
    cast: string[];
    directors: string[];
    firstAirDate: string;
    originCountry: string;
    availabilityStatus: string;
    described: boolean;
    videoTracks: EpisodeVideoTrack[];
    audioTracks: EpisodeAudioTrack[];
    textTracks: EpisodeTextTrack[];
    cuePoints: EpisodeCuePoint[];
};

export type NextEpisode = {
    showID: string;
    seasonNumber: number;
};

export type EpisodeStream = {
    type: string;
    url: string;
};

export type EpisodeVideoTrack = {
    name: string;
    height: number;
    width: number;
    bitrate: number;
};

export type EpisodeAudioTrack = {
    code: string;
    name: string;
    originalName: string;
};

export type EpisodeTextTrack = {
    type: string;
    name: string;
    language: string;
    trackURL: string;
};

export type EpisodeCuePoint = {
    name: string;
    start: number;
    end: number;
};
